src/browser-pool.js

```diff
@@ -195,22 +197,23 @@ class BrowserPool extends EventEmitter {
      * @return {Set<BrowserController>}
      * @private
      */
-    _getAllOpenBrowsers() {
-        return new Set([...this.activeBrowserControllers, ...this.retiredBrowserControllers]);
+    _getAllBrowserControllers() {
+        return Array.from(new Set([...this.activeBrowserControllers, ...this.retiredBrowserControllers]));
```

Comment:
Oh shit. My bad! `Set` has a `forEach` but I refactored it to `map` in `closeAllBrowsers`. I think it would be better to have this function still return a `Set` and instead do 

```js
        const controllers = this._getAllBrowserControllers();
        const promises = [];
        controllers.forEach((controller) => {
            promises.push(controller.close());
        });
```

Because it's more expressive. The line

```js
Array.from(new Set([...this.activeBrowserControllers, ...this.retiredBrowserControllers]));
```

Is a bit hard to understand why this is happening.